mesonbuild/interpreter/primitives/string.py

```diff
@@ -90,12 +91,14 @@ def endswith_method(self, args: T.Tuple[str], kwargs: TYPE_kwargs) -> bool:
     @noArgsFlattening
     @noKwargs
     @typed_pos_args('str.format', varargs=object)
-    def format_method(self, args: T.Tuple[T.List[object]], kwargs: TYPE_kwargs) -> str:
+    def format_method(self, args: T.Tuple[T.List[TYPE_var]], kwargs: TYPE_kwargs) -> str:
         arg_strings: T.List[str] = []
         for arg in args[0]:
-            if isinstance(arg, bool): # Python boolean is upper case.
-                arg = str(arg).lower()
-            arg_strings.append(str(arg))
+            try:
+                arg_strings.append(stringifyUserArguments(arg, self.subproject))
+            except InvalidArguments as e:
+                FeatureBroken.single_use(f'str.format: {str(e)}', '1.3.0', self.subproject)
```

Comment:
Can we pass `location=self.current_node`?

Reply:
done